eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -1032,6 +1033,35 @@ public function unassignRoleFromUser(APIRole $role, User $user)
     }
 
     /**
+     * Removes the given role assignment.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to remove a role assignment
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the role assignment was not found
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleAssignment $roleAssignment
+     */
+    public function unassignRoleByAssignment(RoleAssignment $roleAssignment)
+    {
+        if ($this->repository->canUser('role', 'assign', $roleAssignment) !== true) { //SNAFU
```

Comment:
Work in progress :) It's not ready for a detailed review yet, I mainly need approval of the direction. But thanks of course for suggestions so far, will incorporate them when possible!